src/core/atomic.d

```diff
@@ -119,9 +169,9 @@ version( CoreDdoc )
      * Returns:
      *  The value of 'val'.
      */
-    HeadUnshared!(T) atomicLoad(MemoryOrder ms = MemoryOrder.seq,T)( ref const shared T val ) pure nothrow @nogc @safe
+    TailShared!T atomicLoad(MemoryOrder ms = MemoryOrder.seq,T)( ref const shared T val ) pure nothrow @nogc @safe
```

Comment:
Because `TailShared!T` is more restrictive than `HeadUnshared!T` for classes, we should keep an overload with `HeadUnshared!T` for classes and deprecate it.
Not yet sure how to implement the deprecation though.